server/server.go

```diff
@@ -592,14 +608,12 @@ func (s *server) Health(ctx context.Context, req *rpcpb.HealthRequest) (*rpcpb.H
 		return nil, err
 	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
-	s.clusterInfo.NodeNames = s.network.nodeNames
-	s.clusterInfo.NodeInfos = s.network.nodeInfos
-	s.clusterInfo.Healthy = true
+	info.NodeNames = s.network.nodeNames
```

Comment:
Here too, because there's only a read lock, someone could be writing the info at the same time.